### PR TITLE
fix(migrations): Correctly populate version column

### PR DESCRIPTION
**Pull request by lynnagara**
Ensure the version column in the migrations table is always
incremented when a row is rewritten. This ensures that the
correct row is replaced when duplicate rows are merged.

### CODE REVIEW

File: snuba/migrations/runner.py [hunk 1]
```diff
@@ -91,22 +92,37 @@ def run_migration(self, migration_key: MigrationKey) -> None:
...
         )
 
+    def _get_next_version(self, migration_key: MigrationKey) -> int:
+        group = escape_string(migration_key.group.value)
+        migration_id = escape_string(migration_key.migration_id)
+        conditions = f"group = {group} AND migration_id = {migration_id}"
+
+        result = self.__connection.execute(
+            f"SELECT version FROM {TABLE_NAME} FINAL WHERE {conditions};",
+            settings={"load_balancing": "in_order"},
```
Comment (tkaemming) [inline snuba/migrations/runner.py:118]: It seems like this specific setting gets passed consistently for each query in here — isn't this what the client settings objects are for? Or is this special cased for some other reason?

**Review by tkaemming — APPROVED** [review]

File: snuba/migrations/runner.py (hunk 1 above)
Comment (lynnagara) [reply snuba/migrations/runner.py:118]: Good point, this probably applies anytime we are using the migrate settings, will move into there.

File: snuba/migrations/runner.py [hunk 2]
```diff
@@ -91,22 +92,37 @@ def run_migration(self, migration_key: MigrationKey) -> None:
...
             }
         ]
         self.__connection.execute(
             statement, data, settings={"load_balancing": "in_order"}
         )
 
+    def _get_next_version(self, migration_key: MigrationKey) -> int:
+        group = escape_string(migration_key.group.value)
+        migration_id = escape_string(migration_key.migration_id)
+        conditions = f"group = {group} AND migration_id = {migration_id}"
```
[the remark this answers is not in the document]
Comment (lynnagara) [reply snuba/migrations/runner.py]: i didn't know about this, updated

[the remark this answers is not in the document]
Comment (tkaemming) [reply snuba/migrations/runner.py]: Looks good, thanks

**Pull request merged**